src/python/ensembl/io/genomio/assembly/download.py

```diff
@@ -102,47 +152,59 @@ def get_checksums(checksum_path: Path) -> Dict[str, str]:
     return sums
 
 
-def download_files(accession: str, dl_dir: Path, max_redo: int) -> None:
+# def download_files(accession: str, dl_dir: Path, max_redo: int) -> None:
+def download_files(ftp_connection: FTP, accession: str, dl_dir: Path, max_redo: int) -> None:
     """
     Given an INSDC accession, download all available files from the ftp to the download dir
+
+    Args:
+        ftp_connection: An open FTP connection object
+        accession: Genome assembly accession.
+        dl_dir: Path to downloaded FTP files.
+        max_redo: Maximum ftp connection retry attempts.
+
+    Returns:
+        None
     """
-    match = re.match(r"(GC[AF])_([0-9]{3})([0-9]{3})([0-9]{3})\.?([0-9]+)", accession)
-    if not match:
-        raise UnsupportedFormatError(f"Could not recognize GCA accession format: {accession}")
-    gca = match.group(1)
-    part1 = match.group(2)
-    part2 = match.group(3)
-    part3 = match.group(4)
 
-    # Get the list of assemblies for this accession
-    ftp_url = "ftp.ncbi.nlm.nih.gov"
-    sub_dir = Path("genomes", "all", gca, part1, part2, part3)
-    ftp_conn = FTP()
-    ftp_conn.connect(ftp_url)
-    ftp_conn.login()
-    ftp_conn.cwd(str(sub_dir))
+    # # Get the list of assemblies for this accession
+    for ftp_dir, _ in ftp_connection.mlsd():
+        if re.search(accession, ftp_dir):
+            ftp_connection.cwd(ftp_dir)
 
-    for ftp_dir, _ in ftp_conn.mlsd():
-        if re.match(accession, ftp_dir):
-            ftp_conn.cwd(ftp_dir)
             # First, get the md5sum file
             md5_file = "md5checksums.txt"
             md5_path = dl_dir / md5_file
             with md5_path.open("wb") as fp:
-                ftp_conn.retrbinary(f"RETR {md5_file}", fp.write)
+                ftp_connection.retrbinary(f"RETR {md5_file}", fp.write)
             md5_sums = get_checksums(md5_path)
 
             # Get all the files
-            for ftp_file, _ in ftp_conn.mlsd():
+            for ftp_file, _ in ftp_connection.mlsd():
                 for end in _FILE_ENDS:
                     if ftp_file.endswith(end) and not ftp_file.endswith(f"_from_{end}"):
-                        _download_file(ftp_conn, ftp_file, md5_sums, dl_dir, max_redo)
+                        _download_file(ftp_connection, ftp_file, md5_sums, dl_dir, max_redo)
+        else:
+            logging.warning(
+                f"Could not find accession '{accession}' from ftp {ftp_dir} in open FTP connection"
+            )
 
 
 def _download_file(
-    ftp_conn: FTP, ftp_file: str, md5_sums: Dict[str, str], dl_dir: Path, max_redo: int = 0
+    ftp_connection: FTP, ftp_file: str, md5_sums: Dict[str, str], dl_dir: Path, max_redo: int = 0
 ) -> None:
-    """TODO"""
+    """Download individual files from FTP server.
```

Comment:
```suggestion
    """Downloads individual files from FTP server.
```
Just a suggestion, as it seems the standard is to have descriptions start with a verb in third person :man_shrugging: